### PR TITLE
dev-lang/nim: improvements

### PR DESCRIPTION
**Pull request by CyberTailor**
* respect flags
* add experimental "--depfile" patch

### CODE REVIEW

**Comment by xgqt** [comment]
I'm wondering whether we need `experimental` a flag for the `depile` patch or maybe it should be called `gentoo` rather than experimental. 

@thesamesam what do you think?

File: dev-lang/nim/nim-1.6.6.ebuild [hunk 1]
```diff
@@ -46,6 +34,31 @@ nim_use_enable() {
 	use "${1}" && echo "-d:${2}"
 }
 
+# Borrowed from nim-utils.eclass (guru overlay).
+nim_gen_config() {
+	cat > nim.cfg <<- EOF || die "Failed to create Nim config"
+	cc:"gcc"
```
Comment (xgqt) [inline dev-lang/nim/nim-1.6.6.ebuild:40]: I have a feeling this can backfire. 
Is `gcc.exe:"$(tc-getCC)"` not enough?

Comment (CyberTailor) [reply dev-lang/nim/nim-1.6.6.ebuild:40]: It's only used for choosing what flags to pass AFAIK.

**Comment by xgqt** [comment]
I think `experimental` is not too far off looking at https://packages.gentoo.org/useflags/experimental

**Comment by xgqt** [comment]
@CyberTailor since we change how binaries are compiled - do a revbump :)

**Comment by CyberTailor** [comment]
Done.

**Comment by xgqt** [comment]
Thanks!

**Pull request closed without merging**